build.sbt

```diff
@@ -259,169 +234,320 @@ lazy val coreJVMTests = Project(id = "coreJVMTests", base = file("enumeratum-cor
-    .settings(testSettings: _*)
+    .settings(commonWithPublishSettings)
+    .settings(testSettings)
     .settings(
-      version            := "1.7.2-SNAPSHOT",
+      version            := Versions.Core.head,
       crossScalaVersions := scalaVersionsAll,
-      libraryDependencies ++= Seq(
-        "org.json4s"   %% "json4s-core"   % json4sVersion,
-        "org.json4s"   %% "json4s-native" % json4sVersion % Test,
-        "com.beachape" %% "enumeratum"    % Versions.Core.stable
-      )
+      libraryDependencies ++= {
+        val ver = "4.0.3"
+
+        Seq(
+          "org.json4s" %% "json4s-core"   % ver,
+          "org.json4s" %% "json4s-native" % ver % Test,
+          scalaXmlTest.value
+        )
+      },
+      libraryDependencies ++= {
+        if (useLocalVersion) {
+          Seq.empty
+        } else {
+          Seq("com.beachape" %%% "enumeratum" % Versions.Core.stable)
+        }
+      }
     )
+    .settings(
+      // TODO: Remove once JSON4S is fixed for Scala3;
+      // https://github.com/json4s/json4s/issues/1035
+      disabledSettings
+    )
+    .configure(configureWithLocal(coreJVM))
 
+// ScalaCheck
 lazy val scalacheckAggregate =
   aggregateProject("scalacheck", enumeratumScalacheckJs, enumeratumScalacheckJvm)
 
 lazy val enumeratumScalacheck = crossProject(JSPlatform, JVMPlatform)
   .crossType(CrossType.Pure)
   .in(file("enumeratum-scalacheck"))
-  .settings(commonWithPublishSettings: _*)
-  .settings(testSettings: _*)
-  .jsSettings(jsTestSettings: _*)
+  .settings(commonWithPublishSettings)
+  .settings(testSettings)
+  .jsSettings(jsTestSettings)
   .settings(
     name               := "enumeratum-scalacheck",
-    version            := "1.7.1-SNAPSHOT",
+    version            := Versions.Core.head,
     crossScalaVersions := scalaVersionsAll,
     libraryDependencies ++= {
+      val ver: String = {
+        if (scalaBinaryVersion.value == "2.11") {
+          "1.15.2"
+        } else {
+          "1.15.4"
+        }
+      }
+
       Seq(
-        "com.beachape"      %%% "enumeratum"      % Versions.Core.stable,
-        "org.scalacheck"    %%% "scalacheck"      % theScalacheckVersion(scalaVersion.value),
-        "org.scalatestplus" %%% "scalacheck-1-14" % "3.1.1.1"            % Test,
-        "com.beachape"      %%% "enumeratum-test" % Versions.Core.stable % Test
+        "org.scalacheck"    %%% "scalacheck"      % ver,
+        "org.scalatestplus" %%% "scalacheck-1-14" % "3.1.1.1" % Test
+      ).map(
+        _.exclude("org.scala-lang.modules", "*")
+          .exclude("org.scalatest", "*")
+          .cross(CrossVersion.for3Use2_13)
       )
+    },
+    libraryDependencies += scalaXmlTest.value,
+    libraryDependencies ++= {
+      if (useLocalVersion) {
+        Seq.empty
+      } else {
+        Seq(
+          "com.beachape" %%% "enumeratum"      % Versions.Core.stable,
+          "com.beachape" %%% "enumeratum-test" % Versions.Core.stable % Test
+        )
```

Comment:
This is re-used a few times in the file; anyway to DRY it out?

Reply:
Yes, but anyway a deep refactoring of `build.sbt` would be required, but I would prefer to do it after